src/Passes/Source/RuleTransformationPass/RulePass.cpp

```diff
@@ -0,0 +1,863 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "RuleTransformationPass/RulePass.hpp"
+#include "Rules/Factory.hpp"
+#include "Rules/Notation/Notation.hpp"
+#include "Rules/ReplacementRule.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <fstream>
+#include <iostream>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    RuleTransformationPass::RuleTransformationPass(
+        RuleSet&&                                  rule_set,
+        RuleTransformationPassConfiguration const& config,
+        Profile*                                   profile)
+      : rule_set_{std::move(rule_set)}
+      , config_{config}
+      , profile_{profile}
+    {
+    }
+
+    void RuleTransformationPass::setupCopyAndExpand()
+    {
+        using namespace microsoft::quantum::notation;
+        addConstExprRule(
+            {branch("cond"_cap = constInt(), "if_false"_cap = _, "if_true"_cap = _),
+             [](Builder& builder, Value* val, Captures& captures, Replacements& replacements) {
+                 auto cst   = llvm::dyn_cast<llvm::ConstantInt>(captures["cond"]);
+                 auto instr = llvm::dyn_cast<llvm::Instruction>(val);
+
+                 if (cst == nullptr || instr == nullptr)
+                 {
+                     return false;
+                 }
+
+                 auto branch_cond = cst->getValue().getZExtValue();
+                 auto if_true     = llvm::dyn_cast<llvm::BasicBlock>(captures["if_true"]);
+                 auto if_false    = llvm::dyn_cast<llvm::BasicBlock>(captures["if_false"]);
+
+                 auto type = instr->getType();
+                 if (type == nullptr)
+                 {
+                     return false;
+                 }
+
+                 if (branch_cond)
+                 {
+                     builder.CreateBr(if_true);
+                     instr->replaceAllUsesWith(llvm::UndefValue::get(type));
+                 }
+                 else
+                 {
+                     builder.CreateBr(if_false);
+                     instr->replaceAllUsesWith(llvm::UndefValue::get(type));
+                 }
```

Comment:
Consider placing line 
43 between 51 and 53;
44 between 54 and 55;
45 .. 59 and 60.

Reply:
Thanks.